Makefile

```diff
@@ -0,0 +1,272 @@
+PROJECT_NAME=devopsconsole-operator
+PACKAGE_NAME:=github.com/fabric8-services/$(PROJECT_NAME)
+CUR_DIR=$(shell pwd)
+TMP_PATH=$(CUR_DIR)/tmp
+INSTALL_PREFIX=$(CUR_DIR)/bin
+VENDOR_DIR=vendor
+INCLUDE_DIR=make
+SOURCE_DIR ?= .
+SOURCES := $(shell find $(SOURCE_DIR) -path $(SOURCE_DIR)/vendor -prune -o -name '*.go' -print)
+
+BINARY_SERVER_BIN=$(INSTALL_PREFIX)/devconsole-operator
+GO_BIN_NAME=go
+GIT_BIN_NAME=git
+DEP_BIN_NAME=dep
+DOCKER_BIN_NAME=docker
+UNAME_S=$(shell uname -s)
+GOCOV_BIN=$(VENDOR_DIR)/github.com/axw/gocov/gocov/gocov
+GOCOVMERGE_BIN=$(VENDOR_DIR)/github.com/wadey/gocovmerge/gocovmerge
+GOCYCLO_DIR=$(VENDOR_DIR)/github.com/fzipp/gocyclo
+GOCYCLO_BIN=$(GOCYCLO_DIR)/gocyclo
+GOLANGCI_LINT_BIN_NAME:=golangci-lint
+
+# declares variable that are OS-sensitive
+include ./$(INCLUDE_DIR)/test.mk
+include ./$(INCLUDE_DIR)/Makefile.dev
+
+DOCKER_BIN := $(shell command -v $(DOCKER_BIN_NAME) 2> /dev/null)
+include ./$(INCLUDE_DIR)/docker.mk
+
+# This is a fix for a non-existing user in passwd file when running in a docker
+# container and trying to clone repos of dependencies
+GIT_COMMITTER_NAME ?= "user"
+GIT_COMMITTER_EMAIL ?= "user@example.com"
+export GIT_COMMITTER_NAME
+export GIT_COMMITTER_EMAIL
+
+COMMIT=$(shell git rev-parse HEAD 2>/dev/null)
+GITUNTRACKEDCHANGES := $(shell git status --porcelain --untracked-files=no)
+ifneq ($(GITUNTRACKEDCHANGES),)
+	COMMIT := $(COMMIT)-dirty
+endif
+BUILD_TIME=`date -u '+%Y-%m-%dT%H:%M:%SZ'`
+
+.DEFAULT_GOAL := help
+
+# Call this function with $(call log-info,"Your message")
+define log-info =
+	@echo "INFO: $(1)"
+endef
+
+# -------------------------------------------------------------------
+# Docker build
+# -------------------------------------------------------------------
+BUILD_DIR = bin
+REGISTRY_URI = quay.io
+REGISTRY_NS = ${PROJECT_NAME}
+REGISTRY_IMAGE = ${PROJECT_NAME}
+
+ifeq ($(TARGET),rhel)
+	REGISTRY_URL := ${REGISTRY_URI}/openshiftio/rhel-${REGISTRY_NS}-${REGISTRY_IMAGE}
```

Comment:
can we change docker org as we are starting fresh?

Reply:
Same as below. Let's do this when we know how it will look like.